src/corefx/System.Globalization.Native/collation.cpp

```diff
@@ -126,6 +126,84 @@ extern "C" int32_t LastIndexOf(
 }
 
 /*
+Static Function:
+AreEqualOrdinalIgnoreCase
+*/
+static bool AreEqualOrdinalIgnoreCase(UChar one, UChar two)
+{
+	// Return whether the two characters are identical or would be identical if they were upper-cased.
+
+	if (one == two)
+	{
+		return true;
+	}
+
+	if (one == 0x0131 || two == 0x0131)
+	{
+		// On Windows with InvariantCulture, the LATIN SMALL LETTER DOTLESS I (U+0131)
+		// capitalizes to itself, whereas with ICU it capitalizes to LATIN CAPITAL LETTER I (U+0049).
+		// We special case it to match the Windows invariant behavior.
+		return false;
+	}
+
+	return u_toupper(one) == u_toupper(two);
+}
+
+/*
+Function:
+IndexOfOrdinalIgnoreCase
+*/
+extern "C" int32_t
+IndexOfOrdinalIgnoreCase(const UChar* lpTarget, int32_t cwTargetLength, const UChar* lpSource, int32_t cwSourceLength)
+{
+	int32_t endIndex = cwSourceLength - cwTargetLength;
+	assert(endIndex >= 0);
+
+	for (int32_t i = 0; i <= endIndex; i++)
+	{
+		int32_t targetIdx = 0;
+		for (int32_t srcIdx = i; targetIdx < cwTargetLength; srcIdx++, targetIdx++) {
+			if (!AreEqualOrdinalIgnoreCase(lpSource[srcIdx], lpTarget[targetIdx])) {
```

Comment:
Invariant culture casing (and the ordinal casing) in Windows doesn't support Surrogate casing. I think Windows do that to avoid changing Invariant behavior. but I don't see any reason why we don't support surrogate casing for ordinal. we may diverge from the desktop here but we can support that in desktop though some quirk. I think using en-US casing for surrogate will do the trick. I didn't try it myself but I expect it will just work